package/MDAnalysis/core/selection.py

```diff
@@ -327,9 +345,11 @@ def _apply_KDTree(self, group):
 
     def _apply_distmat(self, group):
         sel = self.sel.apply(group)
-        ref = sel.center_of_geometry().reshape(1, 3).astype(np.float32)
-
-        box = group.dimensions if self.periodic else None
+        periodic = False if self.validate_dimensions(group.dimensions) is None\
```

Comment:
shorter
```python
periodic = self.validate_dimensions(group.dimensions) is not None
```

Reply:
But even better, rearrange your code to avoid calling the same method twice:
```python
box = self.validate_dimensions(group.dimensions)
periodic = box is not None
ref = sel.center_of_geometry(pbc=periodic).reshape(1, 3).astype(np.float32)
```

Reply:
:+1: